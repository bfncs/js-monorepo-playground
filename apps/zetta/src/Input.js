import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Input extends Component {
  static propTypes = {
    onEnter: PropTypes.func.isRequired,
  };

  state = {
    text: '',
  };

  handleChange = e => this.setState({ text: e.target.value });

  handleKeyPress = e => {
    if (e.key === 'Enter') {
      this.props.onEnter(this.state.text);
    }
  };

  render() {
    return (
      <input type="text" value={this.state.text} onChange={this.handleChange} onKeyPress={this.handleKeyPress}/>
    )
  }
}

export default Input;