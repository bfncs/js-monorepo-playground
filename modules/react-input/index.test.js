import React from 'react';
import renderer from 'react-test-renderer';
import Input from './index';

const NO_OP = () => {};

describe('react-input snapshot tests', () => {

  it('matches snapshot when done', () => {
    expect(renderer.create(
      <Input onEnter={NO_OP}/>
    ).toJSON()).toMatchSnapshot();
  });

});