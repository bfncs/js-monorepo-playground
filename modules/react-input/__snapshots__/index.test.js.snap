// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-input snapshot tests matches snapshot when done 1`] = `
<input
  onChange={[Function]}
  onKeyPress={[Function]}
  type="text"
  value=""
/>
`;
